tempesta_fw/sock.c

```diff
@@ -145,6 +145,8 @@ ss_send(struct sock *sk, const SsSkbList *skb_list)
 	/* Synchronize concurrent socket writting in different softirqs. */
 	bh_lock_sock_nested(sk);
 
+	if (sk->sk_state != TCP_ESTABLISHED)
+		goto out;
```

Comment:
We definitely can send data in CLOSE_WAIT state. Probably we also can send in SYN_SENT and/or SYN_RCVD - need to check.